dimod/vartypes.py

```diff
@@ -117,6 +121,16 @@ class Vartype(enum.Enum):
     INTEGER = DISCRETE = Integers()  # DISCRETE is an alias for INTEGER
     REAL = Real()
 
+    # Dev note:
+    # This allows us to treat QMs and BQMs in the same way, i.e. we can
+    # do bqm.vartype(v).
+    # It would be better to make BQM.vartype a method, but that would be an
+    # enormous backward compatibility break.
+    # I tried instead making a proxy object, but that doesn't allow syntax like
+    # bqm.vartype is SPIN
+    def __call__(self, v: typing.Optional[dimod.typing.Variable] = None) -> Vartype:
+        return self
```

Comment:
Should `QM.vartype` impl differ (use `v`)?

Reply:
`QM.vartype` is already just a method that required `v`. Though it does mean a user can do `QM.vartype(v)()` :shrug: .

I tried a lot of more restrictive approaches but unfortunately it's super buttoned down.

Reply:
Oh, got it. Missed it's already a method.